File: ChatGPT/Complexas/Pluck/Pluck.py
def pluck_elements(matrix, elements_to_pluck):
    def is_valid(i, j):
        return 0 <= i < len(matrix) and 0 <= j < len(matrix[0])

    def pluck(i, j):
        if not is_valid(i, j) or matrix[i][j] == 0:
            return
        matrix[i][j] = 0
        for x, y in [(i-1, j), (i, j-1), (i, j), (i, j+1), (i+1, j)]:
            if is_valid(x, y):
                matrix[x][y] = 0

    plucks = 0
    for i, j in elements_to_pluck:
        if matrix[i][j] == 1:
            pluck(i, j)
            plucks += 1
    
    return plucks

# Example usage
matrix = [
    [1, 1, 0],
    [1, 1, 1],
    [0, 1, 0]
]

elements_to_pluck = [(0, 0), (1, 1), (2, 1)]
print("Total plucks:", pluck_elements(matrix, elements_to_pluck))

File: ChatGPT/Complexas/Pluck/Pluck.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;

class MatrixPlucker {
public:
    MatrixPlucker(vector<vector<int>>& mat, const set<pair<int, int>>& elemsToPluck)
        : matrix(mat), elementsToPluck(elemsToPluck), rows(mat.size()), cols(mat[0].size()) {}

    int pluckElements() {
        int plucks = 0;
        for (const auto& elem : elementsToPluck) {
            int i = elem.first;
            int j = elem.second;
            if (matrix[i][j] == 1) {
                pluck(i, j);
                ++plucks;
            }
        }
        return plucks;
    }

private:
    vector<vector<int>>& matrix;
    const set<pair<int, int>>& elementsToPluck;
    int rows, cols;

    void pluck(int i, int j) {
        if (i < 0 || i >= rows || j < 0 || j >= cols || matrix[i][j] == 0) return;
        matrix[i][j] = 0;
        vector<pair<int, int>> positions = {{i-1, j}, {i, j-1}, {i, j}, {i, j+1}, {i+1, j}};
        for (const auto& pos : positions) {
            int x = pos.first, y = pos.second;
            if (x >= 0 && x < rows && y >= 0 && y < cols) {
                matrix[x][y] = 0;
            }
        }
    }
};

// Example usage
int main() {
    vector<vector<int>> matrix = {
        {1, 1, 0},
        {1, 1, 1},
        {0, 1, 0}
    };

    set<pair<int, int>> elementsToPluck = {{0, 0}, {1, 1}, {2, 1}};
    MatrixPlucker plucker(matrix, elementsToPluck);
    cout << "Total plucks: " << plucker.pluckElements() << endl;

    return 0;
}